.changeset/smart-drinks-fold.md

```diff
@@ -0,0 +1,5 @@
+---
+'opticks': minor
+---
+
+:bug: Codemods: Fix variable cleanup that was too aggressive and cleaned optimistically
```

Comment:
Will this automatically bump the version number? Or does that need to be done manually?

Reply:
Yes it will be updated automatically with the changeset action (thanks @dale-french!)

Reply:
> Thanks for the improvements, Jop!
> 
> It's definitely a great improvement. I didn't see any major issues. However, I noticed a few places in our `client` package where the cleanup did not work as expected. Let me know if you think these issues are outside of the scope of this PR. In that case, I can create some new `Issues` on the repo for us to fix those in the future.
> 
> * This unused component declaration
>   
>   * https://github.com/FindHotel/daedalus/blob/d7e94ae04391572630ada85ee33c98f2f31aa7ea/packages/client/src/components/DesktopHeader/index.tsx#L53
> * Unused prop values
>   
>   * https://github.com/FindHotel/daedalus/blob/d7e94ae04391572630ada85ee33c98f2f31aa7ea/packages/client/src/components/DesktopHeader/index.tsx#L127
>   * This is probably because the prop type definition still references the prop name
>   * In any case, this requires more manual clean-up, so maybe it's better that it isn't cleaned up at the moment
> * Unused destructuring consts
>   
>   * https://github.com/FindHotel/daedalus/blob/d7e94ae04391572630ada85ee33c98f2f31aa7ea/packages/client/src/components/Filters/FacetFilter/FacetFilter.tsx#L109
>   * https://github.com/FindHotel/daedalus/blob/d7e94ae04391572630ada85ee33c98f2f31aa7ea/packages/client/src/components/FiltersOverlay/FiltersOverlay.tsx#L21
> * Unused const declaration
>   
>   * https://github.com/FindHotel/daedalus/blob/d7e94ae04391572630ada85ee33c98f2f31aa7ea/packages/client/src/components/FiltersPopovers/FiltersScrollWrapper/ScrollButton/index.tsx#L25

Thanks Gerrit! I think we can consider those for another separate PR, since they won't break the codebase if I understand correctly? If you find a particular pattern to be more abundant than others, let me know and we can take a look how hard it is to implement the clean up for them.